parquet-column/src/main/java/org/apache/parquet/filter2/predicate/Operators.java

```diff
@@ -415,14 +420,18 @@ public int hashCode() {
     }
   }
 
-  private static class ContainsColumnPredicate<T extends Comparable<T>, U extends ColumnFilterPredicate<T>>
+  private static class ContainsColumnPredicate<T extends Comparable<T>, U extends SingleColumnFilterPredicate<T>>
       extends Contains<T> {
     private final U underlying;
 
     ContainsColumnPredicate(U underlying) {
       super(underlying.getColumn());
-      if (underlying.getValue() == null) {
-        throw new IllegalArgumentException("Contains predicate does not support null element value");
+      if ((underlying instanceof ColumnFilterPredicate && ((ColumnFilterPredicate) underlying).getValue() == null)
+          || underlying instanceof SetColumnFilterPredicate
+              && ((SetColumnFilterPredicate) underlying)
+                  .getValues()
+                  .contains(null)) {
```

Comment:
nit: I know it means the same logically, but I think it is more readable to wrap in a parenthesis since the `instanceof` and the related cast are tightly related.

Reply:
makes sense! updated